frontend/src/metabase/dashboard/components/add-card-sidebar/AddCardSidebar/AddCardSidebar.unit.spec.tsx

```diff
@@ -0,0 +1,274 @@
+  ...ROOT,
+  can_write: true,
+});
+
+const COLLECTIONS = [
+  ROOT_COLLECTION,
+  COLLECTION,
+  SUBCOLLECTION,
+  PERSONAL_COLLECTION,
+  PERSONAL_SUBCOLLECTION,
+];
+
+interface SetupOpts {
+  collections: Collection[];
+  collectionItems?: CollectionItem[];
+  dashboard?: Dashboard;
+}
+
+async function setup({
+  collections,
+  collectionItems = [],
+  dashboard = createMockDashboard({
+    collection: ROOT_COLLECTION,
+  }),
+}: SetupOpts) {
+  setupCollectionsEndpoints({
+    collections,
+  });
+  setupCollectionItemsEndpoint({
+    collection: createMockCollection(checkNotNull(dashboard.collection)),
+    collectionItems,
+  });
+
+  renderWithProviders(<AddCardSidebar onSelect={jest.fn()} />, {
+    storeInitialState: createMockState({
+      currentUser: CURRENT_USER,
+      dashboard: createMockDashboardState({
+        dashboards: {
+          [dashboard.id]: { ...dashboard, dashcards: [] },
+        },
+        dashboardId: dashboard.id,
+      }),
+    }),
+  });
+
+  await waitFor(() => {
+    expect(screen.queryByText("Loading...")).not.toBeInTheDocument();
+  });
+}
+
+describe("AddCardSideBar", () => {
+  it("should render no items", async () => {
+    await setup({
+      collections: [],
+    });
+
+    expect(screen.getByText("Nothing here")).toBeInTheDocument();
+  });
+
+  it("should hide all personal collections when adding questions to dashboards in the root collection (public collection)", async () => {
+    const dashboardInPublicCollection = createMockDashboard({
+      collection: ROOT_COLLECTION,
+    });
+    await setup({
+      collections: COLLECTIONS,
+      dashboard: dashboardInPublicCollection,
+    });
+
+    assertBreadcrumbs([ROOT_COLLECTION]);
+
+    expect(
+      screen.getByRole("menuitem", {
+        name: COLLECTION.name,
+      }),
+    ).toBeInTheDocument();
+
+    expect(
+      screen.queryByRole("menuitem", {
+        name: PERSONAL_COLLECTION.name,
+      }),
+    ).not.toBeInTheDocument();
+  });
+
+  it("should show all questions when adding questions to dashboards in the root collection (public collection)", async () => {
+    const dashboardInRootCollection = createMockDashboard({
+      collection: ROOT_COLLECTION,
+    });
+
+    const collectionItems = [
+      createMockCollectionItem({
+        id: getNextId(),
+        model: "card",
+        name: "question 1",
+      }),
+      createMockCollectionItem({
+        id: getNextId(),
+        model: "card",
+        name: "question 2",
+      }),
+    ];
+    await setup({
+      collections: COLLECTIONS,
+      dashboard: dashboardInRootCollection,
+      collectionItems,
+    });
+
+    assertBreadcrumbs([ROOT_COLLECTION]);
+
+    collectionItems.forEach(collectionItem => {
+      expect(
+        screen.getByRole("menuitem", {
+          name: collectionItem.name,
+        }),
+      ).toBeInTheDocument();
+    });
+  });
+
+  it("should show all questions when adding questions to dashboards in public collections", async () => {
```

Comment:
test Dashboards > #1 from [the epic](https://github.com/metabase/metabase/issues/34733)